dm/dm/master/server.go

```diff
@@ -1070,6 +1075,52 @@ func (s *Server) getStatusFromWorkers(ctx context.Context, sources []string, tas
 	}
 	wg.Wait()
 	s.fillUnsyncedStatus(workerResps)
+
+	// find if all sources status are provided in resps
+	findEnoughTaskInResp := func(taskName string, needCnt int) bool {
+		cnt := 0
+		for _, resp := range workerResps {
+			for _, status := range resp.SubTaskStatus {
+				if status.Name == taskName {
+					cnt += 1
+				}
+				if cnt == needCnt {
+					return true
+				}
+			}
+		}
+		return false
+	}
+	appendFakeResp := func(taskName, sourceName, msg string) {
+		setWorkerResp(&pb.QueryStatusResponse{
+			Result: false, Msg: msg, SubTaskStatus: []*pb.SubTaskStatus{{Name: taskName}},
+			SourceStatus: &pb.SourceStatus{Source: sourceName, Worker: "unknown-offline-worker"},
+		})
+	}
+	// when taskName is empty we need list all task even the worker that handle this task is not running.
+	if taskName == "" {
+		for taskName, sourceM := range s.scheduler.GetSubTaskCfgs() {
+			if !findEnoughTaskInResp(taskName, len(sourceM)) {
+				msg := fmt.Sprintf("can't find task: %s from dm-worker, please use dmctl list-member to check if worker is offline.", taskName)
+				log.L().Warn(msg)
+				// only add use specified source related to this task
+				if specifiedSource {
+					for _, needDisplayedSource := range sources {
+						if _, ok := sourceM[needDisplayedSource]; ok && !findEnoughTaskInResp(taskName, len(sourceM)) {
+							appendFakeResp(taskName, needDisplayedSource, msg)
+						}
+					}
+				} else {
+					// make fake response for every source related to this task
+					for source := range sourceM {
+						if !findEnoughTaskInResp(taskName, len(sourceM)) {
```

Comment:
you have called in `findEnoughTaskInResp` in line 1103, please refine you logic

Reply:
addressed in https://github.com/pingcap/ticdc/pull/3358/commits/c14607a407c3fc145cce764dc10e5e00ddea299c